app/controllers/admin/batches_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Admin::BatchesController < ApplicationController
+  before_action :set_admin_batch, only: [:show, :edit, :update, :destroy]
+
+  def new
+    @batch = Batch.new
+    @roast = Roast.find(params[:roast])
+    @batch.roast = Roast.find(params[:roast])
```

Comment:
We can reuse the `@roast` fetched and assign it directly to the `@batch.roast`. 
By calling `Roast.find` twice we are querying the database twice for the same data.
```ruby
@batch.roast = @roast
```